src/sdl2/render.rs

```diff
@@ -492,30 +496,81 @@ impl TextureCreator {
     }
 }
 
+pub enum TargetKind {
+    Canvas,
+    Texture,
+}
+
+pub trait Target {
```

Comment:
Unfortunately we cannot make this a private Trait since the Renderer "exposes" it as part of it's generic parameter type. However, because the fields in the Renderer are private, no one can create their own `Renderer` for their own custom `Target`. I added comments on top of the trait to explain that it is internal use only.